src/effects/builtin/noisegen/noiseviewmodel.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Audacity: A Digital Audio Editor
+ */
+#include "noiseviewmodel.h"
+
+#include "log.h"
+
+using namespace au::effects;
+
+QStringList NoiseViewModel::typeNames() const
+{
+    static const QStringList names {
+        tr("White", "not a color, but 'white noise' having a uniform spectrum"),
+        tr("Pink", "not a color, but 'pink noise' having a spectrum with more power in low frequencies"),
+        tr("Brownian", "a kind of noise spectrum also known as 'red' or 'brown'"),
+    };
+
+    return names;
+}
+
+QVariantMap NoiseViewModel::types() const
+{
+    static const QVariantMap typeNames {
+        {
+            QString::number((int)NoiseEffect::Type::White),
+            tr("White", "not a color, but 'white noise' having a uniform spectrum")
+        },
+        {
+            QString::number((int)NoiseEffect::Type::Pink),
+            tr("Pink", "not a color, but 'pink noise' having a spectrum with more power in low frequencies")
+        },
+        {
+            QString::number((int)NoiseEffect::Type::Brownian),
+            tr("Brownian", "a kind of noise spectrum also known as 'red' or 'brown'")
+        },
+    };
+
+    return typeNames;
+}
+
+double NoiseViewModel::amplitude() const
+{
+    const NoiseEffect* const effect_ = effect();
+    IF_ASSERT_FAILED(effect_) {
+        return 0.0;
+    }
+    return effect_->amplitude();
+}
+
+void NoiseViewModel::setAmplitude(double newAmplitude)
+{
+    NoiseEffect* const effect_ = effect();
+    IF_ASSERT_FAILED(effect_) {
+        return;
+    }
+    effect_->setAmplitude(newAmplitude);
```

Comment:
I recommend doing the implementation as in reverb

Reply:
will do